### PR TITLE
Set namespace and snapshot fields to optional to support older kernels

### PR DESCRIPTION
**Pull request by jlowellwofford**

### CODE REVIEW

File: pkg/krbd/device.go [hunk 1]
```diff
@@ -76,13 +76,16 @@ func (d *Device) readDeviceAttrs(path string) error {
...
-		tag := t.Field(i).Tag.Get("krbd")
-		if tag != "" {
-			r, err := os.Open(path + "/" + tag)
-			defer r.Close()
+		tag := getDeviceTag(t.Field(i))
+		if tag.name != "" {
+			r, err := os.Open(path + "/" + tag.name)
 			if err != nil {
+				if tag.optional {
+					continue
```
Comment (bensallen) [inline pkg/krbd/device.go:84]: Perhaps we should only allow `errors.Is(err, os.ErrNotExist)` if the tag is optional, so down the road we don't miss passing up any odd errors when reading these optional attributes.

Comment (jlowellwofford) [reply pkg/krbd/device.go:84]: Seems like a good idea.

New push uses:
```go
if tag.optional && errors.Is(err, os.ErrNotExist) {
```

**Comment by jlowellwofford** [comment]
We were able to successfully test this on a 3.10 kernel and it worked as expected.

**Comment by bensallen** [comment]
Thanks!

**Pull request merged**